omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,265 @@
+			    getUriOfObject(iterator.getObjectType(), iterator.getObjectUuid()));
+		}
+		description.addSelfLink();
+		description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		if (tagList.size() == 0) {
+			description = null;
+		}
+		return convertDelegateToRepresentation(delegate, description);
+	}
+	
+	@RepHandler(FullRepresentation.class)
+	public SimpleObject asFull(Tag delegate) throws ConversionException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		if (StringUtils.isNotBlank(delegate.getObjectType())) {
+			description.addProperty("uuid");
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addProperty("auditInfo");
+		} else {
+			List<Tag> tags = getService().getTags(delegate.getTag(), true);
+			SimpleObject simpleObject = new SimpleObject();
+			List<Map<String, Object>> taggedObjects = new ArrayList<Map<String, Object>>();
+			for (Tag t : tags) {
+				Map<String, Object> taggedObject = new LinkedHashMap<String, Object>();
+				taggedObject.put("tag", t.getTag());
+				taggedObject.put("uuid", t.getUuid());
+				taggedObject.put("objectType", t.getObjectType());
+				taggedObject.put("objectUuid", t.getObjectUuid());
+				taggedObjects.add(taggedObject);
+			}
+			simpleObject.put("taggedObjects", taggedObjects);
+			return simpleObject;
+		}
+		description.addSelfLink();
+		return convertDelegateToRepresentation(delegate, description);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getRepresentationDescription(Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		return description;
+	}
+	
+	/**
+	 * @return the TagService
+	 */
+	private TagService getService() {
+		return Context.getService(TagService.class);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("tag");
+		description.addRequiredProperty("objectType");
+		description.addRequiredProperty("objectUuid");
+		
+		return description;
+	}
+	
+	/**
+	 * @see DelegatingCrudResource#doGetAll(RequestContext)
+	 */
+	@Override
+	protected NeedsPaging<Tag> doGetAll(RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	private String getUriOfObject(String objectType, String objectUuid) {
+		Class<?> className = null;
+		try {
+			className = Context.loadClass(objectType);
+		}
+		catch (ClassNotFoundException e) {}
+		BaseDelegatingResource resource = (BaseDelegatingResource) Context.getService(RestService.class)
+		        .getResourceBySupportedClass(className);
+		String Uri = resource.getUri(resource.getByUniqueId(objectUuid));
```

Comment:
combine these 2 lines to:
`
return resource.getUri(resource.getByUniqueId(objectUuid));`